.circleci/config.yml

```diff
@@ -1,45 +1,230 @@
-# Ruby CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-ruby/ for more details
-#
-version: 2
-jobs:
-  build:
-    parallelism: 1
-    docker:
-      # specify the version you desire here
-      # https://circleci.com/developer/images/image/cimg/ruby
-      - image: cimg/ruby:3.2.1
-        environment:
-          CODECLIMATE_REPO_TOKEN: b6626e682a8e97e0c5978febc92c3526792a2d018b41b8e1b52689da37fb7d92
-
-    working_directory: ~/knapsack_pro-ruby
+version: 2.1
 
+commands:
+  setup_knapsack_pro_ruby:
     steps:
       - checkout
-
-      # Download and cache dependencies
       - restore_cache:
           keys:
-          - v2-dependencies-bundler-{{ checksum "knapsack_pro.gemspec" }}
-          # fallback to using the latest cache if no exact match is found
-          - v2-dependencies-bundler-
-
+          - v1-bundler-ruby-{{ checksum "knapsack_pro.gemspec" }}
+          - v1-bundler-ruby-
       - run:
-          name: install ruby dependencies
           command: |
-            bundle install --jobs=4 --retry=3 --path vendor/bundle
-
-      - run: gem install rubocop
-
+            bundle config set --local path './vendor/bundle'
+            bundle install --jobs=4 --retry=3
       - save_cache:
           paths:
             - ./vendor/bundle
-          key: v2-dependencies-bundler-{{ checksum "knapsack_pro.gemspec" }}
+          key: v1-bundler-ruby-{{ checksum "knapsack_pro.gemspec" }}
 
-      # enforce `frozen_string_literal: true` in the gem files
-      - run: rubocop -A --only Style/FrozenStringLiteralComment,Layout/EmptyLineAfterMagicComment lib/
+  setup_rails_app_with_knapsack_pro:
+    parameters:
+      path:
+        type: string
+      ruby:
+        type: string
+      rspec:
+        type: string
+    steps:
+      - run:
+          working_directory: << parameters.path >>
+          command: |
+            git clone --depth 1 git@github.com:KnapsackPro/rails-app-with-knapsack_pro.git ./
```

Comment:
We could use the following. This way you could test na new changes on a feature branch. For example you create a `featureA` branch in the `knapsack_pro-ruby` repository. If you run tests then you will use the `master` branch from `rails-app-with-knapsack_pro` (default).

If you need to add some new tests or changes to `rails-app-with-knapsack_pro` then you don't have to push it to the master. You can create the `featureA` branch in the `rails-app-with-knapsack_pro` with your new tests/changes. Then when you run tests for the `featureA` in `knapsack_pro-ruby` then it would use the `featureA` branch from `rails-app-with-knapsack_pro` to test it on CI.  Once you are happy with new tests/changes in `rails-app-with-knapsack_pro` then you can merge `featureA` into `master`. This way you don't have to polluting the `master` branch with tmp code in `rails-app-with-knapsack_pro` until you are sure this code is fine to add to the master.

```suggestion
            git clone --depth 1 --branch $CIRCLE_BRANCH --single-branch git@github.com:KnapsackPro/rails-app-with-knapsack_pro.git ./ || git clone --depth 1 git@github.com:KnapsackPro/rails-app-with-knapsack_pro.git ./
```

We already do something like that [here](https://github.com/KnapsackPro/rails-app-with-knapsack_pro/blob/675a1188f39cf8a67ffb447a4487db50fcf98619/.circleci/config.yml#L46).